library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableByteArrayListMultimapImpl.java

```diff
@@ -152,7 +153,15 @@ public SpillableStateStore getStore()
   @Override
   public List<V> removeAll(@Nullable Object key)
```

Comment:
@brightchen I know you're concerned about the deleted list ending up being garbage, I have the same concern but that does not explain why the unit test is failing.